### PR TITLE
fix: pass the pagination size explicitly in all the search calls

### LINKED ISSUE #5464: Wrong pagination on the search result page if we change the defaultPageSize setting
**Describe the bug**
When we change the [defaultPageSize](https://github.com/plone/volto/blob/main/packages/volto/src/config/index.js#L124) setting, the search result lists 25 results per page, but the pagination follows `defaultPageSize`.

**To Reproduce**

1. Change `defaultPageSize` to 6
2. Create 26 pages with the title "Test"
3. Search for the term "Test"
4. See that 25 results are shown on the result page, but the pagination shows 5 results pages.

**Expected behavior**
Pagination shows 2 result pages.

**Screenshots**
download (9)

### PR DESCRIPTION
**Pull request by EshaanAgg**
Fixes #5464 by passing the configured `defaultPageSize` to each API request made to the `search` endpoint.

### CODE REVIEW

**Comment by EshaanAgg** [comment]
Can anyone tell me where and how I should make the changelog entry?

File: packages/volto/news/5464.bugfix
Comment (stevepiercy) [inline packages/volto/news/5464.bugfix:1]:
```suggestion
Fixed pagination in search results by passing `pageSize` explicitly to all search API calls. @EshaanAgg
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Minor grammar on the change log entry. A core contributor should review. I'll ask for a few.

**Review by stevepiercy — APPROVED** [review]

File: packages/volto/src/components/theme/Search/Search.jsx
Comment (tiberiuichim) [inline packages/volto/src/components/theme/Search/Search.jsx:115]: Switch the order of lines here, otherwise it's impossible to set a different page size

Comment (EshaanAgg) [reply packages/volto/src/components/theme/Search/Search.jsx:115]: Nice catch. Thanks!

File: packages/volto/src/components/theme/Search/Search.jsx
Comment (wesleybl) [reply packages/volto/src/components/theme/Search/Search.jsx:114]: @davisagli I put a `console.log` in the constructor and it printed the `defaultPageSize`. I think it's better this way, because you'll have the code in one place.

Comment (EshaanAgg) [reply packages/volto/src/components/theme/Search/Search.jsx:114]: Should I refactor the code?
If yes, please elaborate on what you meant by `class defaultPageSize`? Isn't it a variable?

Comment (tiberiuichim) [reply packages/volto/src/components/theme/Search/Search.jsx:114]: @EshaanAgg I think @wesleybl wants it as a prop, or to extract it as a static field of the Search class component. Having it as a prop would provide some value. I don't see the other option, "extracting it as a field" as providing a lot of value except by making it more readable? I don't know. It would add a level of indirection.

Comment (EshaanAgg) [reply packages/volto/src/components/theme/Search/Search.jsx]: Got it!

**Review by wesleybl — APPROVED** [review]
LGTM

File: packages/volto/src/components/theme/Search/Search.jsx
Comment (ksuess) [inline packages/volto/src/components/theme/Search/Search.jsx:288]: I think here options should be taken into account.
```
const options = qs.parse(this.props.history.location.search);
…

                      this.props.search.items_total / options.b_size ||
                        this.defaultPageSize,
```

**Review by ksuess — CHANGES_REQUESTED** [review]

File: packages/volto/src/components/theme/Search/Search.jsx
Comment (EshaanAgg) [reply packages/volto/src/components/theme/Search/Search.jsx:128]: Ahh got it. Thanks!

File: packages/volto/src/components/theme/Search/Search.jsx
Comment (EshaanAgg) [reply packages/volto/src/components/theme/Search/Search.jsx]: Made the required changes.

**Review by ksuess — APPROVED** [review]
It's maybe nitpicking. For some elegance, I would replace the multiple `const options = qs.parse(this.props.history.location.search);`with one
`this.options = qs.parse(props.history.location.search);` in constructor code.

**Review by sneridagh — APPROVED** [review]

**Review by davisagli — COMMENTED** [review]
@sneridagh Any reason we haven't merged this?

**Comment by stevepiercy** [comment]
@davisagli no idea why no one has pressed the shiny green Merge button. We don't need to wait for our BDFL (@sneridagh) to approve to merge good work that has been approved by multiple members of the @plone/volto-team. If any member of the @plone/volto-team thinks a PR should be merged, then just do it! @sneridagh has expressed this multiple times. I'm merging it as soon as CI passes, because it's got buttloads of approval, and I don't know diddly squat about React 😁.

**Pull request merged**
